src/liballoc/rc.rs

```diff
@@ -1751,17 +1751,17 @@ impl<T> Weak<T> {
     /// This can be used to safely get a strong reference (by calling [`upgrade`]
     /// later) or to deallocate the weak count by dropping the `Weak<T>`.
     ///
-    /// It takes ownership of one weak count (with the exception of pointers created by [`new`],
+    /// It takes ownership of one weak handle (with the exception of pointers created by [`new`],
     /// as these don't have any corresponding weak count).
     ///
     /// # Safety
     ///
     /// The pointer must have originated from the [`into_raw`]  and must still own its potential
-    /// weak reference count.
+    /// weak reference handle.
     ///
-    /// It is allowed for the strong count to be 0 at the time of calling this, but the weak count
-    /// must be non-zero or the pointer must have originated from a dangling `Weak<T>` (one created
-    /// by [`new`]).
+    /// It is allowed for the strong count to be 0 at the time of calling this. Nevertheless, this
+    /// turns one conceptual „frozen“ `Weak` in the form of a raw pointer to an actual `Weak` and
+    /// therefore must be paired with a previous call to [`into_raw`].
```

Comment:
The "conceptual frozen" piece and how freezing relates to into_raw is not clear to me. The into_raw doc says nothing about frozenness. What distinction is "conceptual frozen" making here?

Reply:
I've tried to use some reader's intuition there, trying to clarify it, not make some further distinction. But it's probably better to just avoid that.